### ISSUE TITLE
Feature request: Do not always open this site in this container if you are opening this site from another container.

### ISSUE THREAD

**Issue by vicelo23** [mozilla/multi-account-containers#2314]
Currently if you add a domain to Always open in this contaienr of a container settings, it's opening this website in the new container, it's the normal behaviour but there is a thing I'm thinking it's missing, if I already open a tab in a container and enter this site with that container, it's creating a new tab with other container and entering that site in this container.

Steps to reproduce current behaviour.

1. Have a container, for example name is Microsoft.
2. Set github.com as always open this site in Microsoft container.
3. Create another container, for example name is Container2.
4. Create a tab in Container2 and enter github.com

After following these steps, you will see it creates a new tab of Microsoft container and opens github.com in this tab. In this issue I suggest to add a option to change this behaviour. Always open this site in Microsoft container IF you open it from a non-container tab, otherwise continue to open this site in current container, so in this case it should open github.com in Container2.

The reason I suggest this is we can have two container, one for signed in browsing and other for anonymous browsing, lack of this feature is forcing us to use one container in ANY condition. We may want to visit github.com with signed in container in most of the times and to do that we can just write github.com to the new tab address bar and it opens this site in Microsoft container, but if we want to visit github.com in Container2 for anonymous browsing sometimes, we can just open a tab in Container2 and enter github.com. Microsoft and github.com are just an example usage of it, this can be Meta and facebook.com, Google and youtube.com or anything.

**Comment by shtrom** [comment]
I think the Open-in-default-container feature and UI should have a slightly different behaviour.

In addition to the behaviour described above, it would be good to have a wider selection of options beyond “Open in default container” or “Make this the container”

I have use cases where I open the same site under multiple accounts, as scoped by container. I end up switching that site's default container all the time, just to be able to open it in my desired container. Moreover, it sometime opens automatically in the other container from what I currently want, and the only way to deal with it is to first create a container tab in the desired container, then browse to the site and interact with the container-selection UI.

I think a better experience would be to allow more than one container to handle a URL, and to offer a list of containers to open a given URL in, say:

* This container / No container (depending on the tab's current context)
  * This container and remember this
* [Personal] (the first container configured for use for this site)
* [Work] (the second container configured for use for this site)
* [Other project] (yet another container configured for use for this site)
* ...

In the case where the current tab is not in a container, and only one container has been recorded to open the URL, then this could still be done automatically without new UI.

Whenever entering a URL, the page would be shown if the current tab container (or lack thereof) does not correspond to any of the configured containers for the site.

This should cater for the original problem, but also replace the “default container” behaviour with an explicitly selectable list of (pre-chosen) target containers rather than a single implicit default.

**Comment by aetha** [comment]
Bumping this request. Quite a few sites I want to be opened in a specific container by default, but only in place of the default container. When I open a site in a container other than its default one, the container switch prompts in each additional child tab are quite grating. So I don't assign default containers when I would otherwise like to. This feature would be a big QoL improvement for me.

**Comment by Paul-Spagnola-Work** [comment]
I added some sites to always open in a specific container. Later a deleted them from the container site list so they would open in default container. However, these sites continue to open in the specific containers they were tied to no matter how many times I delete them from the container's site list (i.e. "Manage site list" from manage container screen).  Seems somewhat related to this issue.

**Comment by basinilya** [comment]
Can we have a checkbox in container settings that would enable this feature for the container? Then the old behavior would not be broken.

**Comment by xip0rt** [comment]
Came here looking for this feature.

Thoughts: in addition to the default "always open in this container," you add an allowlist that lets you open a tab in another container by right clicking an open tab and telling it to force open in that container (from the "open in new container tab" menu)

**Comment by KaXaSA** [comment]
> Came here looking for this feature.
> 
> Thoughts: in addition to the default "always open in this container," you add an allowlist that lets you open a tab in another container by right clicking an open tab and telling it to force open in that container (from the "open in new container tab" menu)

Yeah, this is what I was expecting when I was testing Firefox multi account containers. 
IMO you should be able to:

1. set a site to be opened in a specific container by default
1. at least allow you to bypass the default container rule by:
`right click the tab > open in new container tab > some other container `

**Comment by thomaseizinger** [comment]
This is absolutely needed. I want to separate my work and personal Google accounts with containers. Most of the time I need my work one but I do occasionally want to go into the personal one. The current behaviour makes this impossible because every time I try to log-in to the private one, it moves me to my work Google container.

Edit: It turns out that this kind of works if you _don't_ have a sub-domain registered as "always open-in". I had `accounts.google.com` in there which is why it took me always to the work container.
